proposal/S413/README.rst

```diff
@@ -0,0 +1,122 @@
+algorithms in python. 
+
+Prospective Users
+------------------
+Anyone with an interest in optimization problems. 
+The applications for the algorithm range from bullet
+design to musical composition (as its origin would have it). 
+
+System Architecture
+--------------------
+Harmony Search begins by initializing a Harmony Memory (HM). HM can be saved
+as a matrix of size (N, HMMS), where N is the number of parameters, and HMMS
+is the HM size. 
+
+The User provides an objective function, which is stored as an array of the coefficients. 
+The HM is initialized and entries are fit to the function. Keeping tabs on the worst and
+best fit is important.
+
+On every iteration of the algorithm, a new harmony is created with a probability of 1-HMCR 
+(Harmony Memory Considering Rate). If an existing entry is chosen instead (p=HCMR), then it
+is altered according to a Pitch Rate (PR).
+
+The new harmony is then fit against the worst fitting entry in HM. IF the new harmony is a better
+fit, then HM is overwritten; otherwise, the new harmony is simply discarded.
+
+Harmony Search can be written as a class containing it required elements HM, HMCR, PR, and the methods
+required to produce and fit the harmonies to the objective function.
+The class should also store the best and worst fits in HM.
+
+::
+
+class HarmonySearch
+
+{
+	
+	vector<vector<float>> HM;
+	
+	float PR;
+	
+	float HCMR;
+	
+	vector<float> best; //can consider just saving an index so as not to copy
+	
+	vector<float> worst; //""""
+	
+	vector<float> objectiveFunction;
+	
+	init_HM(HM); //should initialize the HM randomly
+	
+	fit(objectiveFunction, harmony); //should fit a harmony into Objective
+	
+	generate_harmony(HM, HCMR); //should generate new harmonies at every iteration either from scratch or by calling change_pitch
+	
+	compare_fit(new_harmony, worst, HM); //should compare the new harmony and the worst harmony in HM fit
+	
+	change_pitch(HM, PR); // if a new harmony is not made from scratch, one from HM is modified
+
+}
+
+
+This is a bare-bone rough idea of what the class will be.
+There is also the issue of including different variations 
+of the algorithm. I'm yet unsure if it would require
+separate classes (with base as parent).
+
+API Description
+----------------
+The harmony search algorithm class will be written in C++.
+This will be wrapped in python for python use.
+
+The user can also provide an initialized HM in addition to choosing
+what mode of HSA to use.
+
+
+
+Engineering Infrastructure
+---------------------------
+Testing will be done with python unittest.
+
+The Harmony Search algorithm can be tested against other ER optimization algorithms (since they
+have been compared before as well).
+
+
+Schedule
+--------
+
+* Week 1: Write base Harmony Search Class.
+* Week 2: Write python side: input (OF, optional HM...) reading, output (best harmony, add options for HM output...).
+* Week 3: Pybind and initial testing for bare bones class.
```

Comment:
I like the order of C++, Python and then pybind11 and the short period of time to wrap it up.  The short period of iteration should result in good opportunity of completion of the project.